README.md

```diff
@@ -108,24 +119,137 @@ The compiled workflow can be executed via the DNAnexus command line client or we
 
 ![this](doc/bam_chrom_counter.png)
 
-### Strict syntax
+### Validate the workflow
 
 dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that adheres strictly to the WDL specifications. Most of the problematic automatic type conversions that are allowed by some other WDL runtime engines are not allowed by dxCompiler. Please use the command line tools in wdlTools (e.g. `check` and `lint`) to validate your WDL files before trying to compile them with dxCompiler.
 
 ## CWL
 
-To compile a CWL workflow/tool, you will first need to process it using the following steps:
+### Preprocess CWL workflow
+
+dxCompiler requires the source CWL file to be "packed" as a cwl.json file, which contains a single compound workflow with all the dependent processes included. Additionally, you may need to upgrade the version of your workflow to 1.2. The following examples demonstrate how to do that:
 
-1. Upgrade it to CWL 1.2 if it is not already: 
+1. Install `cwl-upgrader` and upgrade your workflow to CWL 1.2 (if it is not already): 
     ```
-    $ cwl-upgrader my-workflow.cwl
+    $ pip install cwl-upgrader
+    $ cwl-upgrader my-workflow.cwl [subworkflow1.cwl subworkflow2.cwl ...]
```

Comment:
1. do all subworkflows mentioned by my-workflow.cwl have to be included on the command line?
2. does cwl-upgrader modify the input files in-place?

Reply:
1. Yes. Although after packing only the wf version is kept https://github.com/rabix/sbpack/blob/6abf31ce7085c071c26a8a55462d3257023e9662/sbpack/pack.py#L247-L250
but to make sure the syntax in subworkflows also works it is better to upgrade all dependent cwl files.
2. if we are in the folder of the cwl files, then it is in-place modification. I added another suggestion if the user want to save the upgraded ones to a new folder